### ISSUE TITLE
Consider adding a CHANGELOG 

### ISSUE THREAD

**Issue by devonhk** [external-secrets/external-secrets#3119]
**Is your feature request related to a problem? Please describe.**

Please consider adding and maintaining a CHANGELOG file as part of your release process.
It's currently difficult to find breaking changes when performing pre-upgrade checks.

The process appears to be the following:
- Cut a release
- Collect PR titles since last release and add them as bullet points in the new [release page](https://github.com/external-secrets/external-secrets/releases/tag/v0.9.0)

My current review process is to quickly look through each commit to look for breaking changes.

**Describe the solution you'd like**

The [k8s](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.26.md#urgent-upgrade-notes) project has great changelogs. I don't know if ESO needs something as formal, but it would be nice to have a changelog with high level summaries of what was changed and what isn't backwards compatible. This of course requires maintainers/release leads to manually write descriptions instead of relying on commit messages/PR titles.

**Comment by finalsecure** [comment]
ne diyon be abi gözünü sevim be abla

**Comment by finalsecure** [comment]
### ### ### ### ### ### ### 

> `_**YARRRAK**_`

**Comment by finalsecure** [comment]
### a

**Comment by finalsecure** [comment]
azqwxsecdrvftghjkzaqwxsedcrfvtbgynumıöozqawxsecdrvtybunjımköoçpazqwxsecrvtbynujmıköoçpazwxsecdrvtbynumıözxcymxcrfvtgbyhnujmı4bnmöbnmködcvgybhujıkögyh jmkövgybhnjmkögybhnjımkdlgvjnersfgıbonorpfgnoakdrenmgoknareogmşlrafemvbşlmareıgh9ıarewıognjodrzhgıuargnlrıughajegnlrenışoughea5rgnlkaedrlkgaothgokamtpihjgoığagoıaretolgaojhgıjnaktabnaıjerthgnbokakşghrıjaogmlkamfbaıuneghmalkmbrokjaonbamklrgokaer`5lrkjgjnkargkjn qkglnbkjabgrçömnglakjgbrşagnalrjvkşanvlkajrgf ### ### ### ### **_kvar`_** qergqgorhqıgkjnbqıurhgıqjngpqoerjgknşelkjhgpaıwngşakejrgfohreaıgnarekngkşaRNGIUhgrınKLGN`wefjbıwrgqrgnqırgıqergnkqrıgbkqnregıhqreg` `جمعة مباركة`

**Comment by finalsecure** [comment]
`بلا أم`

**Comment by finalsecure** [comment]
![Ekran görüntüsü 2024-02-07 011741](https://github.com/external-secrets/external-secrets/assets/84764425/5f57d476-930e-4c5b-9943-efdad1f4f57e)

**Comment by finalsecure** [comment]
![Uploading Ekran Görüntüsü (1).png…]()

**Comment by finalsecure** [comment]
![Ekran Görüntüsü (1)](https://github.com/external-secrets/external-secrets/assets/84764425/c4858165-d54b-4e22-b59a-4edb2b5d04a2)

**Comment by finalsecure** [comment]
![indir](https://github.com/external-secrets/external-secrets/assets/84764425/deb4212f-58a0-4eff-8b5c-0df876f2cd09)

**Comment by finalsecure** [comment]
![10181884](https://github.com/external-secrets/external-secrets/assets/84764425/056aa0d6-ef62-48ec-a581-584f6eae35f3)

**Comment by matheusmazzoni** [comment]
Can someone assing to me? :)
I belive we can create a changelog file using as the terraform modules are released:
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/.github/workflows/release.yml
What you think guys?

**Comment by Skarlso** [comment]
Yep, LGTM.

**Comment by matheusmazzoni** [comment]
I am working on this, but I would like to take advantage of the discussion and implement the automatic release system as well, that is, when a PR is committed on the main, depending on its title, it will generate a major/minior/patch as well as the terraform module repo I showed above. Do you think they make sense, @Skarlso ?

**Comment by Skarlso** [comment]
You mean major minor patch in the changelog, or what? :D

**Comment by evrardj-roche** [comment]
I do not believe that automatically crafted changelog will help in any way. I believe that hand crafted upgrade release notes would help however. I think the work need to be done in policy.

Do you want to tackle that @matheusmazzoni ?

**Comment by matheusmazzoni** [comment]
I got too busy and didn't pay any more attention to the project.

So the ideia, is the ESO releases would be generated based on commits made in main, using conventional commits for this. So, as we have today in some terraforms modules: 
- https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/.github/workflows/release.yml
- https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/CHANGELOG.md
- https://github.com/terraform-aws-modules/terraform-aws-eks/releases?page=2

We do not take advantage of commit messages to generate both the desired version and automatic changelogs that have been modified.

**Comment by Skarlso** [comment]
Instead of duplicating information and adding something plus to maintain I suggest a simpler approach.

Whatever pr is a breaking change( if we even know beforehand that it is a breaking change ) we mark with a `!` that will automatically label it with breaking change label.

After that it's trivial to parse any pull requests with a breaking change label into its own section in the release notes that we generate for each release.

**Comment by matheusmazzoni** [comment]
Hi! I've opened a draft PR addressing this issue: https://github.com/external-secrets/external-secrets/pull/6292                                                                                     

Regarding testing the release workflow step: since the "Update CHANGELOG" step uses GITHUB_TOKEN with contents: write and pushes to main, it can only be fully tested by a maintainer triggering the release workflow on the upstream repo. However, the core logic (fetching release notes via gh api and prepending to the file) can be validated manually by any maintainer running the shell commands locally against an existing release tag, without needing to trigger the full workflow.

**Comment by matheusmazzoni** [comment]
Hey everyone,

Following up on the discussion we just had over at [PR #6292](https://github.com/external-secrets/external-secrets/pull/6292#issuecomment-4472106119), I'd like to bring the conversation back here to align on the actual scope of this issue.

Initially, I was leaning towards introducing some automation for the changelog process. However, based on the feedback, it feels like that might be out of scope or not the priority right now.

Could we clarify what the exact goal is here? Are we looking for a purely documentation-focused update, or is there a specific direction we want to take regarding the changelog structure?

Once we lock down the requirements, I’ll be happy to either adjust the current PR or open a new one that fits the exact scope. Thanks!
